### PR TITLE
Fix generate-stackbrew-library.sh to work on macos

### PR DESCRIPTION
**Pull request by janhoy**
Fixed for macos. Depends on GNU version of `find` so had to install that as well :) 

This should help with #278 on Macos

### CODE REVIEW

File: update.md [hunk 1]
```diff
@@ -99,15 +99,16 @@ Using [Homebrew](https://brew.sh/), install the necessary dependencies for macOS
...
-# Make gnu readlink the default. You may wish to undo this after building 
+# Make gnu readlink and find the default. You may wish to undo this after building 
 ln -s /usr/local/bin/greadlink /usr/local/bin/readlink
+ln -s /usr/local/bin/gfind /usr/local/bin/find
 ```
 
-NOTE: If you don't want to symlink readlink permanently you may instead
-create a symbolic link to `/path/to/some/bin/readlink` and put that location
+NOTE: If you don't want to symlink readlink and gfind permanently you may instead
+create symbolic links to `/path/to/some/bin/readlink` and `/path/to/somr/bin/find` and put that location
```
Comment (makuk66) [inline update.md]: `some`

Comment (makuk66) [inline update.md]: `some`

**Comment by dsmiley** [comment]
Instead of symlink-ing in "find", can/should the bash script refer to "gfind" instead?  I modified it as so and it worked.  Perhaps "gfind" isn't referred to as such on a linux host but only in Mac OS with brew?  I don't know.

**Comment by janhoy** [comment]
We could have a check at the start of each main script to check for correct versions of tools and tell the user to run `. macos_init.sh` which would do the required setup for this shell only.

**Comment by dsmiley** [comment]
That works for me... it's a shame there are these troublesome platform differences.  I like this "this shell only" aspect here; I don't want to symlink core executable utilities in my system if I can help it.

**Comment by janhoy** [comment]
The last commit adds an init script that can be sourced to put a folder with symlinked GNU tools first in path. User needs to source this in each shell before starting to run commands...

File: tools/init_macos.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+hash greadlink 2>/dev/null || { echo >&2 "Required tool greadlink is not installed, please run 'brew install coreutils'"; exit 1; }
+hash gfind 2>/dev/null || { echo >&2 "Required tool gfind is not installed, please run 'brew install findutils'"; exit 1; }
+if [[ ! -d /tmp/docker-solr-bin ]]; then
+  mkdir -p /tmp/docker-solr-bin >/dev/null 2>&1
+  ln -s /usr/local/bin/greadlink /tmp/docker-solr-bin/readlink
+  ln -s /usr/local/bin/gfind /tmp/docker-solr-bin/find
+fi
+if [[ ! $PATH == *"docker-solr-bin"* ]]; then
+  export PATH=/tmp/docker-solr-bin:$PATH
+  echo "Configuring for macOS - ádding /tmp/docker-solr-bin first in path"
```
Comment (dsmiley) [inline tools/init_macos.sh]: Why the accent on the 'a' in "adding"?

Comment (janhoy) [reply tools/init_macos.sh]: Oops :)

**Pull request merged**